// viem/dist/accounts/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _chunk4EJIKB23js = require('../chunk-4EJIKB23.js');









var _chunk2DDJFGFEjs = require('../chunk-2DDJFGFE.js');

// src/accounts/index.ts
var _bip32 = require('@scure/bip32');

// src/accounts/generateMnemonic.ts
var _bip39 = require('@scure/bip39');
function generateMnemonic(wordlist10) {
  return _bip39.generateMnemonic.call(void 0, wordlist10);
}

// src/accounts/generatePrivateKey.ts
var _secp256k1 = require('@noble/curves/secp256k1');
function generatePrivateKey() {
  return _chunk2DDJFGFEjs.toHex.call(void 0, _secp256k1.secp256k1.utils.randomPrivateKey());
}

// src/accounts/privateKeyToAccount.ts


// src/accounts/utils/sign.ts

async function sign({
  hash,
  privateKey
}) {
  const { r, s, recovery } = _secp256k1.secp256k1.sign(hash.slice(2), privateKey.slice(2));
  return {
    r: _chunk2DDJFGFEjs.toHex.call(void 0, r),
    s: _chunk2DDJFGFEjs.toHex.call(void 0, s),
    v: recovery ? 28n : 27n
  };
}

// src/accounts/utils/signatureToHex.ts

function signatureToHex({ r, s, v }) {
  return `0x${new _secp256k1.secp256k1.Signature(
    _chunk2DDJFGFEjs.hexToBigInt.call(void 0, r),
    _chunk2DDJFGFEjs.hexToBigInt.call(void 0, s)
  ).toCompactHex()}${_chunk2DDJFGFEjs.toHex.call(void 0, v).slice(2)}`;
}

// src/accounts/utils/signMessage.ts
async function signMessage({
  message,
  privateKey
}) {
  const signature = await sign({ hash: _chunk2DDJFGFEjs.hashMessage.call(void 0, message), privateKey });
  return signatureToHex(signature);
}

// src/accounts/utils/signTransaction.ts
async function signTransaction({
  privateKey,
  transaction
}) {
  const signature = await sign({
    hash: _chunk2DDJFGFEjs.keccak256.call(void 0, _chunk2DDJFGFEjs.serializeTransaction.call(void 0, transaction)),
    privateKey
  });
  return _chunk2DDJFGFEjs.serializeTransaction.call(void 0, transaction, signature);
}

// src/accounts/utils/signTypedData.ts
async function signTypedData({
  privateKey,
  ...typedData
}) {
  const signature = await sign({
    hash: _chunk2DDJFGFEjs.hashTypedData.call(void 0, typedData),
    privateKey
  });
  return signatureToHex(signature);
}

// src/accounts/privateKeyToAccount.ts
function privateKeyToAccount(privateKey) {
  const publicKey = _chunk2DDJFGFEjs.toHex.call(void 0, _secp256k1.secp256k1.getPublicKey(privateKey.slice(2), false));
  const address = _chunk2DDJFGFEjs.publicKeyToAddress.call(void 0, publicKey);
  const account = _chunk4EJIKB23js.toAccount.call(void 0, {
    address,
    async signMessage({ message }) {
      return signMessage({ message, privateKey });
    },
    async signTransaction(transaction) {
      return signTransaction({ privateKey, transaction });
    },
    async signTypedData(typedData) {
      return signTypedData({ ...typedData, privateKey });
    }
  });
  return {
    ...account,
    publicKey,
    source: "privateKey"
  };
}

// src/accounts/hdKeyToAccount.ts
function hdKeyToAccount(hdKey_, { accountIndex = 0, addressIndex = 0, changeIndex = 0, path } = {}) {
  const hdKey = hdKey_.derive(
    path || `m/44'/60'/${accountIndex}'/${changeIndex}/${addressIndex}`
  );
  const account = privateKeyToAccount(_chunk2DDJFGFEjs.toHex.call(void 0, hdKey.privateKey));
  return {
    ...account,
    getHdKey: () => hdKey,
    source: "hd"
  };
}

// src/accounts/mnemonicToAccount.ts


function mnemonicToAccount(mnemonic, opts = {}) {
  const seed = _bip39.mnemonicToSeedSync.call(void 0, mnemonic);
  return hdKeyToAccount(_bip32.HDKey.fromMasterSeed(seed), opts);
}

// src/accounts/wordlists.ts
var _czech = require('@scure/bip39/wordlists/czech');
var _english = require('@scure/bip39/wordlists/english');
var _french = require('@scure/bip39/wordlists/french');
var _italian = require('@scure/bip39/wordlists/italian');
var _japanese = require('@scure/bip39/wordlists/japanese');
var _korean = require('@scure/bip39/wordlists/korean');
var _simplifiedchinese = require('@scure/bip39/wordlists/simplified-chinese');
var _spanish = require('@scure/bip39/wordlists/spanish');
var _traditionalchinese = require('@scure/bip39/wordlists/traditional-chinese');





















exports.HDKey = _bip32.HDKey; exports.czech = _czech.wordlist; exports.english = _english.wordlist; exports.french = _french.wordlist; exports.generateMnemonic = generateMnemonic; exports.generatePrivateKey = generatePrivateKey; exports.hdKeyToAccount = hdKeyToAccount; exports.italian = _italian.wordlist; exports.japanese = _japanese.wordlist; exports.korean = _korean.wordlist; exports.mnemonicToAccount = mnemonicToAccount; exports.parseAccount = _chunk2DDJFGFEjs.parseAccount; exports.privateKeyToAccount = privateKeyToAccount; exports.publicKeyToAddress = _chunk2DDJFGFEjs.publicKeyToAddress; exports.signMessage = signMessage; exports.signTypedData = signTypedData; exports.simplifiedChinese = _simplifiedchinese.wordlist; exports.spanish = _spanish.wordlist; exports.toAccount = _chunk4EJIKB23js.toAccount; exports.traditionalChinese = _traditionalchinese.wordlist;
