// viem/dist/accounts/index.mjs
import {
  toAccount
} from "../chunk-YCI363JZ.mjs";
import {
  hashMessage,
  hashTypedData,
  hexToBigInt,
  keccak256,
  parseAccount,
  publicKeyToAddress,
  serializeTransaction,
  toHex
} from "../chunk-LEU2ZGMI.mjs";

// src/accounts/index.ts
import { HDKey as HDKey2 } from "@scure/bip32";

// src/accounts/generateMnemonic.ts
import { generateMnemonic as generateMnemonic_ } from "@scure/bip39";
function generateMnemonic(wordlist10) {
  return generateMnemonic_(wordlist10);
}

// src/accounts/generatePrivateKey.ts
import { secp256k1 } from "@noble/curves/secp256k1";
function generatePrivateKey() {
  return toHex(secp256k1.utils.randomPrivateKey());
}

// src/accounts/privateKeyToAccount.ts
import { secp256k1 as secp256k14 } from "@noble/curves/secp256k1";

// src/accounts/utils/sign.ts
import { secp256k1 as secp256k12 } from "@noble/curves/secp256k1";
async function sign({
  hash,
  privateKey
}) {
  const { r, s, recovery } = secp256k12.sign(hash.slice(2), privateKey.slice(2));
  return {
    r: toHex(r),
    s: toHex(s),
    v: recovery ? 28n : 27n
  };
}

// src/accounts/utils/signatureToHex.ts
import { secp256k1 as secp256k13 } from "@noble/curves/secp256k1";
function signatureToHex({ r, s, v }) {
  return `0x${new secp256k13.Signature(
    hexToBigInt(r),
    hexToBigInt(s)
  ).toCompactHex()}${toHex(v).slice(2)}`;
}

// src/accounts/utils/signMessage.ts
async function signMessage({
  message,
  privateKey
}) {
  const signature = await sign({ hash: hashMessage(message), privateKey });
  return signatureToHex(signature);
}

// src/accounts/utils/signTransaction.ts
async function signTransaction({
  privateKey,
  transaction
}) {
  const signature = await sign({
    hash: keccak256(serializeTransaction(transaction)),
    privateKey
  });
  return serializeTransaction(transaction, signature);
}

// src/accounts/utils/signTypedData.ts
async function signTypedData({
  privateKey,
  ...typedData
}) {
  const signature = await sign({
    hash: hashTypedData(typedData),
    privateKey
  });
  return signatureToHex(signature);
}

// src/accounts/privateKeyToAccount.ts
function privateKeyToAccount(privateKey) {
  const publicKey = toHex(secp256k14.getPublicKey(privateKey.slice(2), false));
  const address = publicKeyToAddress(publicKey);
  const account = toAccount({
    address,
    async signMessage({ message }) {
      return signMessage({ message, privateKey });
    },
    async signTransaction(transaction) {
      return signTransaction({ privateKey, transaction });
    },
    async signTypedData(typedData) {
      return signTypedData({ ...typedData, privateKey });
    }
  });
  return {
    ...account,
    publicKey,
    source: "privateKey"
  };
}

// src/accounts/hdKeyToAccount.ts
function hdKeyToAccount(hdKey_, { accountIndex = 0, addressIndex = 0, changeIndex = 0, path } = {}) {
  const hdKey = hdKey_.derive(
    path || `m/44'/60'/${accountIndex}'/${changeIndex}/${addressIndex}`
  );
  const account = privateKeyToAccount(toHex(hdKey.privateKey));
  return {
    ...account,
    getHdKey: () => hdKey,
    source: "hd"
  };
}

// src/accounts/mnemonicToAccount.ts
import { HDKey } from "@scure/bip32";
import { mnemonicToSeedSync } from "@scure/bip39";
function mnemonicToAccount(mnemonic, opts = {}) {
  const seed = mnemonicToSeedSync(mnemonic);
  return hdKeyToAccount(HDKey.fromMasterSeed(seed), opts);
}

// src/accounts/wordlists.ts
import { wordlist } from "@scure/bip39/wordlists/czech";
import { wordlist as wordlist2 } from "@scure/bip39/wordlists/english";
import { wordlist as wordlist3 } from "@scure/bip39/wordlists/french";
import { wordlist as wordlist4 } from "@scure/bip39/wordlists/italian";
import { wordlist as wordlist5 } from "@scure/bip39/wordlists/japanese";
import { wordlist as wordlist6 } from "@scure/bip39/wordlists/korean";
import { wordlist as wordlist7 } from "@scure/bip39/wordlists/simplified-chinese";
import { wordlist as wordlist8 } from "@scure/bip39/wordlists/spanish";
import { wordlist as wordlist9 } from "@scure/bip39/wordlists/traditional-chinese";
export {
  HDKey2 as HDKey,
  wordlist as czech,
  wordlist2 as english,
  wordlist3 as french,
  generateMnemonic,
  generatePrivateKey,
  hdKeyToAccount,
  wordlist4 as italian,
  wordlist5 as japanese,
  wordlist6 as korean,
  mnemonicToAccount,
  parseAccount,
  privateKeyToAccount,
  publicKeyToAddress,
  signMessage,
  signTypedData,
  wordlist7 as simplifiedChinese,
  wordlist8 as spanish,
  toAccount,
  wordlist9 as traditionalChinese
};
