// viem/dist/ethers.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _nullishCoalesce(lhs, rhsFn) { if (lhs != null) { return lhs; } else { return rhsFn(); } }

var _chunk4EJIKB23js = require('./chunk-4EJIKB23.js');


var _chunk2DDJFGFEjs = require('./chunk-2DDJFGFE.js');

// src/adapters/ethers.ts
var ethersWalletToAccount = (wallet) => _chunk4EJIKB23js.toAccount.call(void 0, {
  address: wallet.address,
  async signMessage({ message }) {
    return await wallet.signMessage(_chunk2DDJFGFEjs.toBytes.call(void 0, message));
  },
  async signTransaction(txn) {
    return await wallet.signTransaction({
      ...txn,
      gasLimit: txn.gas
    });
  },
  async signTypedData({ domain, types: types_, message }) {
    const { EIP712Domain: _, ...types } = types_;
    const signTypedData = wallet.signTypedData ? wallet.signTypedData.bind(wallet) : wallet._signTypedData.bind(wallet);
    return await signTypedData(
      _nullishCoalesce(domain, () => ( {})),
      types,
      message
    );
  }
});


exports.ethersWalletToAccount = ethersWalletToAccount;
