// viem/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _nullishCoalesce(lhs, rhsFn) { if (lhs != null) { return lhs; } else { return rhsFn(); } } function _optionalChain(ops) { let lastAccessLHS = undefined; let value = ops[0]; let i = 1; while (i < ops.length) { const op = ops[i]; const fn = ops[i + 1]; i += 2; if ((op === 'optionalAccess' || op === 'optionalCall') && value == null) { return undefined; } if (op === 'access' || op === 'optionalAccess') { lastAccessLHS = value; value = fn(value); } else if (op === 'call' || op === 'optionalCall') { value = fn((...args) => value.call(lastAccessLHS, ...args)); lastAccessLHS = undefined; } } return value; }










































































































































































































































































var _chunk2DDJFGFEjs = require('./chunk-2DDJFGFE.js');

// src/clients/transports/createTransport.ts
function createTransport({
  key,
  name,
  request,
  retryCount = 3,
  retryDelay = 150,
  timeout,
  type
}, value) {
  return {
    config: { key, name, request, retryCount, retryDelay, timeout, type },
    request: _chunk2DDJFGFEjs.buildRequest.call(void 0, request, { retryCount, retryDelay }),
    value
  };
}

// src/clients/transports/custom.ts
function custom(provider, config = {}) {
  const { key = "custom", name = "Custom Provider", retryDelay } = config;
  return ({ retryCount: defaultRetryCount }) => createTransport({
    key,
    name,
    request: provider.request.bind(provider),
    retryCount: _nullishCoalesce(config.retryCount, () => ( defaultRetryCount)),
    retryDelay,
    type: "custom"
  });
}

// src/clients/transports/fallback.ts
function fallback(transports_, config = {}) {
  const {
    key = "fallback",
    name = "Fallback",
    rank = true,
    retryCount,
    retryDelay
  } = config;
  return ({ chain, pollingInterval = 4e3, timeout }) => {
    let transports = transports_;
    const transport = createTransport(
      {
        key,
        name,
        async request({ method, params }) {
          const fetch = async (i = 0) => {
            const transport2 = transports[i]({ chain, retryCount: 0, timeout });
            try {
              return await transport2.request({
                method,
                params
              });
            } catch (err) {
              if (_chunk2DDJFGFEjs.isDeterministicError.call(void 0, err))
                throw err;
              if (i === transports.length - 1)
                throw err;
              return fetch(i + 1);
            }
          };
          return fetch();
        },
        retryCount,
        retryDelay,
        type: "fallback"
      },
      {
        transports: transports.map(
          (fn) => fn({ chain, retryCount: 0 })
        )
      }
    );
    if (rank) {
      const rankOptions = typeof rank === "object" ? rank : {};
      rankTransports({
        chain,
        interval: _nullishCoalesce(rankOptions.interval, () => ( pollingInterval)),
        onTransports: (transports_2) => transports = transports_2,
        sampleCount: rankOptions.sampleCount,
        timeout: rankOptions.timeout,
        transports,
        weights: rankOptions.weights
      });
    }
    return transport;
  };
}
function rankTransports({
  chain,
  interval = 4e3,
  onTransports,
  sampleCount = 10,
  timeout = 1e3,
  transports,
  weights = {}
}) {
  const { stability: stabilityWeight = 0.7, latency: latencyWeight = 0.3 } = weights;
  const samples = [];
  const rankTransports_ = async () => {
    const sample = await Promise.all(
      transports.map(async (transport) => {
        const transport_ = transport({ chain, retryCount: 0, timeout });
        const start = Date.now();
        let end;
        let success;
        try {
          await transport_.request({ method: "net_listening" });
          success = 1;
        } catch (e) {
          success = 0;
        } finally {
          end = Date.now();
        }
        const latency = end - start;
        return { latency, success };
      })
    );
    samples.push(sample);
    if (samples.length > sampleCount)
      samples.shift();
    const maxLatency = Math.max(
      ...samples.map(
        (sample2) => Math.max(...sample2.map(({ latency }) => latency))
      )
    );
    const scores = transports.map((_, i) => {
      const latencies = samples.map((sample2) => sample2[i].latency);
      const meanLatency = latencies.reduce((acc, latency) => acc + latency, 0) / latencies.length;
      const latencyScore = 1 - meanLatency / maxLatency;
      const successes = samples.map((sample2) => sample2[i].success);
      const stabilityScore = successes.reduce((acc, success) => acc + success, 0) / successes.length;
      if (stabilityScore === 0)
        return [0, i];
      return [
        latencyWeight * latencyScore + stabilityWeight * stabilityScore,
        i
      ];
    }).sort((a, b) => b[0] - a[0]);
    onTransports(scores.map(([, i]) => transports[i]));
    await _chunk2DDJFGFEjs.wait.call(void 0, interval);
    rankTransports_();
  };
  rankTransports_();
}

// src/clients/transports/http.ts
function http(url, config = {}) {
  const {
    fetchOptions,
    key = "http",
    name = "HTTP JSON-RPC",
    retryDelay
  } = config;
  return ({ chain, retryCount: retryCount_, timeout: timeout_ }) => {
    const retryCount = _nullishCoalesce(config.retryCount, () => ( retryCount_));
    const timeout = _nullishCoalesce(_nullishCoalesce(timeout_, () => ( config.timeout)), () => ( 1e4));
    const url_ = url || _optionalChain([chain, 'optionalAccess', _2 => _2.rpcUrls, 'access', _3 => _3.default, 'access', _4 => _4.http, 'access', _5 => _5[0]]);
    if (!url_)
      throw new (0, _chunk2DDJFGFEjs.UrlRequiredError)();
    return createTransport(
      {
        key,
        name,
        async request({ method, params }) {
          const { result } = await _chunk2DDJFGFEjs.rpc.http(url_, {
            body: {
              method,
              params
            },
            fetchOptions,
            timeout
          });
          return result;
        },
        retryCount,
        retryDelay,
        timeout,
        type: "http"
      },
      {
        url
      }
    );
  };
}

// src/clients/transports/webSocket.ts
function webSocket(url, config = {}) {
  const { key = "webSocket", name = "WebSocket JSON-RPC", retryDelay } = config;
  return ({ chain, retryCount: retryCount_, timeout: timeout_ }) => {
    const retryCount = _nullishCoalesce(config.retryCount, () => ( retryCount_));
    const timeout = _nullishCoalesce(_nullishCoalesce(timeout_, () => ( config.timeout)), () => ( 1e4));
    const url_ = url || _optionalChain([chain, 'optionalAccess', _6 => _6.rpcUrls, 'access', _7 => _7.default, 'access', _8 => _8.webSocket, 'optionalAccess', _9 => _9[0]]);
    if (!url_)
      throw new (0, _chunk2DDJFGFEjs.UrlRequiredError)();
    return createTransport(
      {
        key,
        name,
        async request({ method, params }) {
          const socket = await _chunk2DDJFGFEjs.getSocket.call(void 0, url_);
          const { result } = await _chunk2DDJFGFEjs.rpc.webSocketAsync(socket, {
            body: { method, params },
            timeout
          });
          return result;
        },
        retryCount,
        retryDelay,
        timeout,
        type: "webSocket"
      },
      {
        getSocket() {
          return _chunk2DDJFGFEjs.getSocket.call(void 0, url_);
        },
        async subscribe({ params, onData, onError }) {
          const socket = await _chunk2DDJFGFEjs.getSocket.call(void 0, url_);
          const { result: subscriptionId } = await new Promise(
            (resolve, reject) => _chunk2DDJFGFEjs.rpc.webSocket(socket, {
              body: {
                method: "eth_subscribe",
                params
              },
              onData: (data) => {
                if (typeof data.id === "number") {
                  resolve(data);
                  return;
                }
                if (data.method !== "eth_subscription")
                  return;
                onData(data.params);
              },
              onError: (error) => {
                reject(error);
                _optionalChain([onError, 'optionalCall', _10 => _10(error)]);
              }
            })
          );
          return {
            subscriptionId,
            async unsubscribe() {
              return new Promise(
                (resolve, reject) => _chunk2DDJFGFEjs.rpc.webSocket(socket, {
                  body: {
                    method: "eth_unsubscribe",
                    params: [subscriptionId]
                  },
                  onData: resolve,
                  onError: reject
                })
              );
            }
          };
        }
      }
    );
  };
}

// src/utils/uid.ts
var size2 = 256;
var index = size2;
var buffer;
function uid(length = 11) {
  if (!buffer || index + length > size2 * 2) {
    buffer = "";
    index = 0;
    for (let i = 0; i < size2; i++) {
      buffer += (256 + Math.random() * 256 | 0).toString(16).substring(1);
    }
  }
  return buffer.substring(index, index++ + length);
}

// src/clients/createClient.ts
function createClient({
  chain,
  key = "base",
  name = "Base Client",
  pollingInterval = 4e3,
  transport,
  type = "base"
}) {
  const { config, request, value } = transport({ chain, pollingInterval });
  return {
    chain,
    key,
    name,
    pollingInterval,
    request,
    transport: { ...config, ...value },
    type,
    uid: uid()
  };
}

// src/clients/decorators/public.ts
var publicActions = (client) => ({
  call: (args) => _chunk2DDJFGFEjs.call.call(void 0, client, args),
  createBlockFilter: () => _chunk2DDJFGFEjs.createBlockFilter.call(void 0, client),
  createContractEventFilter: (args) => _chunk2DDJFGFEjs.createContractEventFilter.call(void 0, client, args),
  createEventFilter: (args) => _chunk2DDJFGFEjs.createEventFilter.call(void 0, client, args),
  createPendingTransactionFilter: () => _chunk2DDJFGFEjs.createPendingTransactionFilter.call(void 0, client),
  estimateContractGas: (args) => _chunk2DDJFGFEjs.estimateContractGas.call(void 0, client, args),
  estimateGas: (args) => _chunk2DDJFGFEjs.estimateGas.call(void 0, client, args),
  getBalance: (args) => _chunk2DDJFGFEjs.getBalance.call(void 0, client, args),
  getBlock: (args) => _chunk2DDJFGFEjs.getBlock.call(void 0, client, args),
  getBlockNumber: (args) => _chunk2DDJFGFEjs.getBlockNumber.call(void 0, client, args),
  getBlockTransactionCount: (args) => _chunk2DDJFGFEjs.getBlockTransactionCount.call(void 0, client, args),
  getBytecode: (args) => _chunk2DDJFGFEjs.getBytecode.call(void 0, client, args),
  getChainId: () => _chunk2DDJFGFEjs.getChainId.call(void 0, client),
  getEnsAddress: (args) => _chunk2DDJFGFEjs.getEnsAddress.call(void 0, client, args),
  getEnsAvatar: (args) => _chunk2DDJFGFEjs.getEnsAvatar.call(void 0, client, args),
  getEnsName: (args) => _chunk2DDJFGFEjs.getEnsName.call(void 0, client, args),
  getEnsResolver: (args) => _chunk2DDJFGFEjs.getEnsResolver.call(void 0, client, args),
  getEnsText: (args) => _chunk2DDJFGFEjs.getEnsText.call(void 0, client, args),
  getFeeHistory: (args) => _chunk2DDJFGFEjs.getFeeHistory.call(void 0, client, args),
  getFilterChanges: (args) => _chunk2DDJFGFEjs.getFilterChanges.call(void 0, client, args),
  getFilterLogs: (args) => _chunk2DDJFGFEjs.getFilterLogs.call(void 0, client, args),
  getGasPrice: () => _chunk2DDJFGFEjs.getGasPrice.call(void 0, client),
  getLogs: (args) => _chunk2DDJFGFEjs.getLogs.call(void 0, client, args),
  getStorageAt: (args) => _chunk2DDJFGFEjs.getStorageAt.call(void 0, client, args),
  getTransaction: (args) => _chunk2DDJFGFEjs.getTransaction.call(void 0, client, args),
  getTransactionConfirmations: (args) => _chunk2DDJFGFEjs.getTransactionConfirmations.call(void 0, client, args),
  getTransactionCount: (args) => _chunk2DDJFGFEjs.getTransactionCount.call(void 0, client, args),
  getTransactionReceipt: (args) => _chunk2DDJFGFEjs.getTransactionReceipt.call(void 0, client, args),
  multicall: (args) => _chunk2DDJFGFEjs.multicall.call(void 0, client, args),
  readContract: (args) => _chunk2DDJFGFEjs.readContract.call(void 0, client, args),
  simulateContract: (args) => _chunk2DDJFGFEjs.simulateContract.call(void 0, client, args),
  uninstallFilter: (args) => _chunk2DDJFGFEjs.uninstallFilter.call(void 0, client, args),
  waitForTransactionReceipt: (args) => _chunk2DDJFGFEjs.waitForTransactionReceipt.call(void 0, client, args),
  watchBlocks: (args) => _chunk2DDJFGFEjs.watchBlocks.call(void 0, client, args),
  watchBlockNumber: (args) => _chunk2DDJFGFEjs.watchBlockNumber.call(void 0, client, args),
  watchContractEvent: (args) => _chunk2DDJFGFEjs.watchContractEvent.call(void 0, client, args),
  watchEvent: (args) => _chunk2DDJFGFEjs.watchEvent.call(void 0, client, args),
  watchPendingTransactions: (args) => _chunk2DDJFGFEjs.watchPendingTransactions.call(void 0, client, args)
});

// src/clients/decorators/test.ts
function testActions(client) {
  return {
    dropTransaction: (args) => _chunk2DDJFGFEjs.dropTransaction.call(void 0, client, args),
    getAutomine: () => _chunk2DDJFGFEjs.getAutomine.call(void 0, client),
    getTxpoolContent: () => _chunk2DDJFGFEjs.getTxpoolContent.call(void 0, client),
    getTxpoolStatus: () => _chunk2DDJFGFEjs.getTxpoolStatus.call(void 0, client),
    impersonateAccount: (args) => _chunk2DDJFGFEjs.impersonateAccount.call(void 0, client, args),
    increaseTime: (args) => _chunk2DDJFGFEjs.increaseTime.call(void 0, client, args),
    inspectTxpool: () => _chunk2DDJFGFEjs.inspectTxpool.call(void 0, client),
    mine: (args) => _chunk2DDJFGFEjs.mine.call(void 0, client, args),
    removeBlockTimestampInterval: () => _chunk2DDJFGFEjs.removeBlockTimestampInterval.call(void 0, client),
    reset: (args) => _chunk2DDJFGFEjs.reset.call(void 0, client, args),
    revert: (args) => _chunk2DDJFGFEjs.revert.call(void 0, client, args),
    sendUnsignedTransaction: (args) => _chunk2DDJFGFEjs.sendUnsignedTransaction.call(void 0, client, args),
    setAutomine: (args) => _chunk2DDJFGFEjs.setAutomine.call(void 0, client, args),
    setBalance: (args) => _chunk2DDJFGFEjs.setBalance.call(void 0, client, args),
    setBlockGasLimit: (args) => _chunk2DDJFGFEjs.setBlockGasLimit.call(void 0, client, args),
    setBlockTimestampInterval: (args) => _chunk2DDJFGFEjs.setBlockTimestampInterval.call(void 0, client, args),
    setCode: (args) => _chunk2DDJFGFEjs.setCode.call(void 0, client, args),
    setCoinbase: (args) => _chunk2DDJFGFEjs.setCoinbase.call(void 0, client, args),
    setIntervalMining: (args) => _chunk2DDJFGFEjs.setIntervalMining.call(void 0, client, args),
    setLoggingEnabled: (args) => _chunk2DDJFGFEjs.setLoggingEnabled.call(void 0, client, args),
    setMinGasPrice: (args) => _chunk2DDJFGFEjs.setMinGasPrice.call(void 0, client, args),
    setNextBlockBaseFeePerGas: (args) => _chunk2DDJFGFEjs.setNextBlockBaseFeePerGas.call(void 0, client, args),
    setNextBlockTimestamp: (args) => _chunk2DDJFGFEjs.setNextBlockTimestamp.call(void 0, client, args),
    setNonce: (args) => _chunk2DDJFGFEjs.setNonce.call(void 0, client, args),
    setRpcUrl: (args) => _chunk2DDJFGFEjs.setRpcUrl.call(void 0, client, args),
    setStorageAt: (args) => _chunk2DDJFGFEjs.setStorageAt.call(void 0, client, args),
    snapshot: () => _chunk2DDJFGFEjs.snapshot.call(void 0, client),
    stopImpersonatingAccount: (args) => _chunk2DDJFGFEjs.stopImpersonatingAccount.call(void 0, client, args)
  };
}

// src/clients/decorators/wallet.ts
var walletActions = (client) => ({
  addChain: (args) => _chunk2DDJFGFEjs.addChain.call(void 0, client, args),
  deployContract: (args) => _chunk2DDJFGFEjs.deployContract.call(void 0, client, args),
  getAddresses: () => _chunk2DDJFGFEjs.getAddresses.call(void 0, client),
  getChainId: () => _chunk2DDJFGFEjs.getChainId.call(void 0, client),
  getPermissions: () => _chunk2DDJFGFEjs.getPermissions.call(void 0, client),
  requestAddresses: () => _chunk2DDJFGFEjs.requestAddresses.call(void 0, client),
  requestPermissions: (args) => _chunk2DDJFGFEjs.requestPermissions.call(void 0, client, args),
  sendTransaction: (args) => _chunk2DDJFGFEjs.sendTransaction.call(void 0, client, args),
  signMessage: (args) => _chunk2DDJFGFEjs.signMessage.call(void 0, client, args),
  signTypedData: (args) => _chunk2DDJFGFEjs.signTypedData.call(void 0, client, args),
  switchChain: (args) => _chunk2DDJFGFEjs.switchChain.call(void 0, client, args),
  watchAsset: (args) => _chunk2DDJFGFEjs.watchAsset.call(void 0, client, args),
  writeContract: (args) => _chunk2DDJFGFEjs.writeContract.call(void 0, client, args)
});

// src/clients/createPublicClient.ts
function createPublicClient({
  chain,
  key = "public",
  name = "Public Client",
  transport,
  pollingInterval
}) {
  const client = createClient({
    chain,
    key,
    name,
    pollingInterval,
    transport,
    type: "publicClient"
  });
  return {
    ...client,
    ...publicActions(client)
  };
}

// src/clients/createTestClient.ts
function createTestClient({
  chain,
  key = "test",
  name = "Test Client",
  mode,
  pollingInterval,
  transport
}) {
  const client = {
    ...createClient({
      chain,
      key,
      name,
      pollingInterval,
      transport,
      type: "testClient"
    }),
    mode
  };
  return {
    ...client,
    ...testActions(client)
  };
}

// src/clients/createWalletClient.ts
function createWalletClient({
  account,
  chain,
  transport,
  key = "wallet",
  name = "Wallet Client",
  pollingInterval
}) {
  const client = {
    ...createClient({
      chain,
      key,
      name,
      pollingInterval,
      transport: () => transport({ retryCount: 0 }),
      type: "walletClient"
    }),
    account: account ? _chunk2DDJFGFEjs.parseAccount.call(void 0, account) : void 0
  };
  return {
    ...client,
    ...walletActions(client)
  };
}
































































































































































































exports.AbiConstructorNotFoundError = _chunk2DDJFGFEjs.AbiConstructorNotFoundError; exports.AbiConstructorParamsNotFoundError = _chunk2DDJFGFEjs.AbiConstructorParamsNotFoundError; exports.AbiDecodingDataSizeInvalidError = _chunk2DDJFGFEjs.AbiDecodingDataSizeInvalidError; exports.AbiDecodingZeroDataError = _chunk2DDJFGFEjs.AbiDecodingZeroDataError; exports.AbiEncodingArrayLengthMismatchError = _chunk2DDJFGFEjs.AbiEncodingArrayLengthMismatchError; exports.AbiEncodingLengthMismatchError = _chunk2DDJFGFEjs.AbiEncodingLengthMismatchError; exports.AbiErrorInputsNotFoundError = _chunk2DDJFGFEjs.AbiErrorInputsNotFoundError; exports.AbiErrorNotFoundError = _chunk2DDJFGFEjs.AbiErrorNotFoundError; exports.AbiErrorSignatureNotFoundError = _chunk2DDJFGFEjs.AbiErrorSignatureNotFoundError; exports.AbiEventNotFoundError = _chunk2DDJFGFEjs.AbiEventNotFoundError; exports.AbiEventSignatureEmptyTopicsError = _chunk2DDJFGFEjs.AbiEventSignatureEmptyTopicsError; exports.AbiEventSignatureNotFoundError = _chunk2DDJFGFEjs.AbiEventSignatureNotFoundError; exports.AbiFunctionNotFoundError = _chunk2DDJFGFEjs.AbiFunctionNotFoundError; exports.AbiFunctionOutputsNotFoundError = _chunk2DDJFGFEjs.AbiFunctionOutputsNotFoundError; exports.AbiFunctionSignatureNotFoundError = _chunk2DDJFGFEjs.AbiFunctionSignatureNotFoundError; exports.BaseError = _chunk2DDJFGFEjs.BaseError; exports.BlockNotFoundError = _chunk2DDJFGFEjs.BlockNotFoundError; exports.CallExecutionError = _chunk2DDJFGFEjs.CallExecutionError; exports.ChainDoesNotSupportContract = _chunk2DDJFGFEjs.ChainDoesNotSupportContract; exports.ContractFunctionExecutionError = _chunk2DDJFGFEjs.ContractFunctionExecutionError; exports.ContractFunctionRevertedError = _chunk2DDJFGFEjs.ContractFunctionRevertedError; exports.ContractFunctionZeroDataError = _chunk2DDJFGFEjs.ContractFunctionZeroDataError; exports.DataLengthTooLongError = _chunk2DDJFGFEjs.DataLengthTooLongError; exports.DataLengthTooShortError = _chunk2DDJFGFEjs.DataLengthTooShortError; exports.DecodeLogTopicsMismatch = _chunk2DDJFGFEjs.DecodeLogTopicsMismatch; exports.EnsAvatarUriResolutionError = _chunk2DDJFGFEjs.EnsAvatarUriResolutionError; exports.EstimateGasExecutionError = _chunk2DDJFGFEjs.EstimateGasExecutionError; exports.ExecutionRevertedError = _chunk2DDJFGFEjs.ExecutionRevertedError; exports.FeeCapTooHighError = _chunk2DDJFGFEjs.FeeCapTooHighError; exports.FeeCapTooLowError = _chunk2DDJFGFEjs.FeeCapTooLowError; exports.FilterTypeNotSupportedError = _chunk2DDJFGFEjs.FilterTypeNotSupportedError; exports.HttpRequestError = _chunk2DDJFGFEjs.HttpRequestError; exports.InsufficientFundsError = _chunk2DDJFGFEjs.InsufficientFundsError; exports.InternalRpcError = _chunk2DDJFGFEjs.InternalRpcError; exports.IntrinsicGasTooHighError = _chunk2DDJFGFEjs.IntrinsicGasTooHighError; exports.IntrinsicGasTooLowError = _chunk2DDJFGFEjs.IntrinsicGasTooLowError; exports.InvalidAbiDecodingTypeError = _chunk2DDJFGFEjs.InvalidAbiDecodingTypeError; exports.InvalidAbiEncodingTypeError = _chunk2DDJFGFEjs.InvalidAbiEncodingTypeError; exports.InvalidAddressError = _chunk2DDJFGFEjs.InvalidAddressError; exports.InvalidArrayError = _chunk2DDJFGFEjs.InvalidArrayError; exports.InvalidBytesBooleanError = _chunk2DDJFGFEjs.InvalidBytesBooleanError; exports.InvalidChainIdError = _chunk2DDJFGFEjs.InvalidChainIdError; exports.InvalidDefinitionTypeError = _chunk2DDJFGFEjs.InvalidDefinitionTypeError; exports.InvalidHexBooleanError = _chunk2DDJFGFEjs.InvalidHexBooleanError; exports.InvalidHexValueError = _chunk2DDJFGFEjs.InvalidHexValueError; exports.InvalidInputRpcError = _chunk2DDJFGFEjs.InvalidInputRpcError; exports.InvalidLegacyVError = _chunk2DDJFGFEjs.InvalidLegacyVError; exports.InvalidParamsRpcError = _chunk2DDJFGFEjs.InvalidParamsRpcError; exports.InvalidRequestRpcError = _chunk2DDJFGFEjs.InvalidRequestRpcError; exports.JsonRpcVersionUnsupportedError = _chunk2DDJFGFEjs.JsonRpcVersionUnsupportedError; exports.LimitExceededRpcError = _chunk2DDJFGFEjs.LimitExceededRpcError; exports.MethodNotFoundRpcError = _chunk2DDJFGFEjs.MethodNotFoundRpcError; exports.MethodNotSupportedRpcError = _chunk2DDJFGFEjs.MethodNotSupportedRpcError; exports.NonceMaxValueError = _chunk2DDJFGFEjs.NonceMaxValueError; exports.NonceTooHighError = _chunk2DDJFGFEjs.NonceTooHighError; exports.NonceTooLowError = _chunk2DDJFGFEjs.NonceTooLowError; exports.OffsetOutOfBoundsError = _chunk2DDJFGFEjs.OffsetOutOfBoundsError; exports.ParseRpcError = _chunk2DDJFGFEjs.ParseRpcError; exports.RawContractError = _chunk2DDJFGFEjs.RawContractError; exports.RequestError = _chunk2DDJFGFEjs.RequestError; exports.ResourceNotFoundRpcError = _chunk2DDJFGFEjs.ResourceNotFoundRpcError; exports.ResourceUnavailableRpcError = _chunk2DDJFGFEjs.ResourceUnavailableRpcError; exports.RpcError = _chunk2DDJFGFEjs.RpcError; exports.RpcRequestError = _chunk2DDJFGFEjs.RpcRequestError; exports.SizeExceedsPaddingSizeError = _chunk2DDJFGFEjs.SizeExceedsPaddingSizeError; exports.SwitchChainError = _chunk2DDJFGFEjs.SwitchChainError; exports.TimeoutError = _chunk2DDJFGFEjs.TimeoutError; exports.TipAboveFeeCapError = _chunk2DDJFGFEjs.TipAboveFeeCapError; exports.TransactionExecutionError = _chunk2DDJFGFEjs.TransactionExecutionError; exports.TransactionNotFoundError = _chunk2DDJFGFEjs.TransactionNotFoundError; exports.TransactionReceiptNotFoundError = _chunk2DDJFGFEjs.TransactionReceiptNotFoundError; exports.TransactionRejectedRpcError = _chunk2DDJFGFEjs.TransactionRejectedRpcError; exports.TransactionTypeNotSupportedError = _chunk2DDJFGFEjs.TransactionTypeNotSupportedError; exports.UnknownNodeError = _chunk2DDJFGFEjs.UnknownNodeError; exports.UnknownRpcError = _chunk2DDJFGFEjs.UnknownRpcError; exports.UrlRequiredError = _chunk2DDJFGFEjs.UrlRequiredError; exports.UserRejectedRequestError = _chunk2DDJFGFEjs.UserRejectedRequestError; exports.WaitForTransactionReceiptTimeoutError = _chunk2DDJFGFEjs.WaitForTransactionReceiptTimeoutError; exports.WebSocketRequestError = _chunk2DDJFGFEjs.WebSocketRequestError; exports.assertRequest = _chunk2DDJFGFEjs.assertRequest; exports.assertTransactionEIP1559 = _chunk2DDJFGFEjs.assertTransactionEIP1559; exports.assertTransactionEIP2930 = _chunk2DDJFGFEjs.assertTransactionEIP2930; exports.assertTransactionLegacy = _chunk2DDJFGFEjs.assertTransactionLegacy; exports.boolToBytes = _chunk2DDJFGFEjs.boolToBytes; exports.boolToHex = _chunk2DDJFGFEjs.boolToHex; exports.bytesToBigint = _chunk2DDJFGFEjs.bytesToBigint; exports.bytesToBool = _chunk2DDJFGFEjs.bytesToBool; exports.bytesToHex = _chunk2DDJFGFEjs.bytesToHex; exports.bytesToNumber = _chunk2DDJFGFEjs.bytesToNumber; exports.bytesToString = _chunk2DDJFGFEjs.bytesToString; exports.concat = _chunk2DDJFGFEjs.concat; exports.concatBytes = _chunk2DDJFGFEjs.concatBytes; exports.concatHex = _chunk2DDJFGFEjs.concatHex; exports.createClient = createClient; exports.createPublicClient = createPublicClient; exports.createTestClient = createTestClient; exports.createTransport = createTransport; exports.createWalletClient = createWalletClient; exports.custom = custom; exports.decodeAbiParameters = _chunk2DDJFGFEjs.decodeAbiParameters; exports.decodeErrorResult = _chunk2DDJFGFEjs.decodeErrorResult; exports.decodeEventLog = _chunk2DDJFGFEjs.decodeEventLog; exports.decodeFunctionData = _chunk2DDJFGFEjs.decodeFunctionData; exports.decodeFunctionResult = _chunk2DDJFGFEjs.decodeFunctionResult; exports.defineBlock = _chunk2DDJFGFEjs.defineBlock; exports.defineChain = _chunk2DDJFGFEjs.defineChain; exports.defineTransaction = _chunk2DDJFGFEjs.defineTransaction; exports.defineTransactionReceipt = _chunk2DDJFGFEjs.defineTransactionReceipt; exports.defineTransactionRequest = _chunk2DDJFGFEjs.defineTransactionRequest; exports.encodeAbiParameters = _chunk2DDJFGFEjs.encodeAbiParameters; exports.encodeDeployData = _chunk2DDJFGFEjs.encodeDeployData; exports.encodeErrorResult = _chunk2DDJFGFEjs.encodeErrorResult; exports.encodeEventTopics = _chunk2DDJFGFEjs.encodeEventTopics; exports.encodeFunctionData = _chunk2DDJFGFEjs.encodeFunctionData; exports.encodeFunctionResult = _chunk2DDJFGFEjs.encodeFunctionResult; exports.encodePacked = _chunk2DDJFGFEjs.encodePacked; exports.etherUnits = _chunk2DDJFGFEjs.etherUnits; exports.fallback = fallback; exports.formatBlock = _chunk2DDJFGFEjs.formatBlock; exports.formatEther = _chunk2DDJFGFEjs.formatEther; exports.formatGwei = _chunk2DDJFGFEjs.formatGwei; exports.formatTransaction = _chunk2DDJFGFEjs.formatTransaction; exports.formatTransactionRequest = _chunk2DDJFGFEjs.formatTransactionRequest; exports.formatUnits = _chunk2DDJFGFEjs.formatUnits; exports.fromBytes = _chunk2DDJFGFEjs.fromBytes; exports.fromHex = _chunk2DDJFGFEjs.fromHex; exports.fromRlp = _chunk2DDJFGFEjs.fromRlp; exports.getAbiItem = _chunk2DDJFGFEjs.getAbiItem; exports.getAddress = _chunk2DDJFGFEjs.getAddress; exports.getContract = _chunk2DDJFGFEjs.getContract; exports.getContractAddress = _chunk2DDJFGFEjs.getContractAddress; exports.getContractError = _chunk2DDJFGFEjs.getContractError; exports.getCreate2Address = _chunk2DDJFGFEjs.getCreate2Address; exports.getCreateAddress = _chunk2DDJFGFEjs.getCreateAddress; exports.getEventSelector = _chunk2DDJFGFEjs.getEventSelector; exports.getFunctionSelector = _chunk2DDJFGFEjs.getFunctionSelector; exports.getSerializedTransactionType = _chunk2DDJFGFEjs.getSerializedTransactionType; exports.getTransactionType = _chunk2DDJFGFEjs.getTransactionType; exports.gweiUnits = _chunk2DDJFGFEjs.gweiUnits; exports.hashMessage = _chunk2DDJFGFEjs.hashMessage; exports.hashTypedData = _chunk2DDJFGFEjs.hashTypedData; exports.hexToBigInt = _chunk2DDJFGFEjs.hexToBigInt; exports.hexToBool = _chunk2DDJFGFEjs.hexToBool; exports.hexToBytes = _chunk2DDJFGFEjs.hexToBytes; exports.hexToNumber = _chunk2DDJFGFEjs.hexToNumber; exports.hexToString = _chunk2DDJFGFEjs.hexToString; exports.http = http; exports.isAddress = _chunk2DDJFGFEjs.isAddress; exports.isAddressEqual = _chunk2DDJFGFEjs.isAddressEqual; exports.isBytes = _chunk2DDJFGFEjs.isBytes; exports.isHash = _chunk2DDJFGFEjs.isHash; exports.isHex = _chunk2DDJFGFEjs.isHex; exports.keccak256 = _chunk2DDJFGFEjs.keccak256; exports.labelhash = _chunk2DDJFGFEjs.labelhash; exports.multicall3Abi = _chunk2DDJFGFEjs.multicall3Abi; exports.namehash = _chunk2DDJFGFEjs.namehash; exports.numberToBytes = _chunk2DDJFGFEjs.numberToBytes; exports.numberToHex = _chunk2DDJFGFEjs.numberToHex; exports.pad = _chunk2DDJFGFEjs.pad; exports.padBytes = _chunk2DDJFGFEjs.padBytes; exports.padHex = _chunk2DDJFGFEjs.padHex; exports.parseAbi = _chunk2DDJFGFEjs.parseAbi; exports.parseAbiItem = _chunk2DDJFGFEjs.parseAbiItem; exports.parseAbiParameter = _chunk2DDJFGFEjs.parseAbiParameter; exports.parseAbiParameters = _chunk2DDJFGFEjs.parseAbiParameters; exports.parseEther = _chunk2DDJFGFEjs.parseEther; exports.parseGwei = _chunk2DDJFGFEjs.parseGwei; exports.parseTransaction = _chunk2DDJFGFEjs.parseTransaction; exports.parseUnits = _chunk2DDJFGFEjs.parseUnits; exports.prepareRequest = _chunk2DDJFGFEjs.prepareRequest; exports.recoverAddress = _chunk2DDJFGFEjs.recoverAddress; exports.recoverMessageAddress = _chunk2DDJFGFEjs.recoverMessageAddress; exports.recoverTypedDataAddress = _chunk2DDJFGFEjs.recoverTypedDataAddress; exports.serializeTransaction = _chunk2DDJFGFEjs.serializeTransaction; exports.size = _chunk2DDJFGFEjs.size; exports.slice = _chunk2DDJFGFEjs.slice; exports.sliceBytes = _chunk2DDJFGFEjs.sliceBytes; exports.sliceHex = _chunk2DDJFGFEjs.sliceHex; exports.stringToBytes = _chunk2DDJFGFEjs.stringToBytes; exports.stringToHex = _chunk2DDJFGFEjs.stringToHex; exports.stringify = _chunk2DDJFGFEjs.stringify; exports.toBytes = _chunk2DDJFGFEjs.toBytes; exports.toHex = _chunk2DDJFGFEjs.toHex; exports.toRlp = _chunk2DDJFGFEjs.toRlp; exports.transactionType = _chunk2DDJFGFEjs.transactionType; exports.trim = _chunk2DDJFGFEjs.trim; exports.validateTypedData = _chunk2DDJFGFEjs.validateTypedData; exports.verifyMessage = _chunk2DDJFGFEjs.verifyMessage; exports.verifyTypedData = _chunk2DDJFGFEjs.verifyTypedData; exports.webSocket = webSocket; exports.weiUnits = _chunk2DDJFGFEjs.weiUnits;
